historical_system_profiles/archiver.py

```diff
@@ -125,33 +125,35 @@ def _archive_profile(data, ptc, logger, notification_service):
         # After the new hsp is saved, we need to check for any reason to alert via
         # triggering a notification, i.e. if drift from any associated baselines has
         # occurred.
-        _check_and_send_notifications(
-            inventory_id,
-            account,
-            org_id,
-            host["updated"],
-            host["display_name"],
-            tags,
-            notification_service,
-            service_auth_key,
-            logger,
+        _further_processing(
+            inventory_id=inventory_id,
+            account_id=account,
+            org_id=org_id,
+            system_check_in=host["updated"],
+            display_name=host["display_name"],
+            tags=tags,
+            groups=groups,
+            notification_service=notification_service,
+            service_auth_key=service_auth_key,
+            logger=logger,
         )
 
 
 def _check_if_notification_enabled(baseline_id, service_auth_key, logger):
     return fetch_baselines([baseline_id], service_auth_key, logger)[0]["notifications_enabled"]
 
 
-def _check_and_send_notifications(
-    inventory_id,
-    account_id,
-    org_id,
-    system_check_in,
-    display_name,
-    tags,
-    notification_service,
-    service_auth_key,
-    logger,
+def _further_processing(
+    inventory_id=None,
+    account_id=None,
+    org_id=None,
```

Comment:
I'd recommend to have these three mandatory. That can be done by not specifying their default values.